docs/spec/enums.rst

```diff
@@ -0,0 +1,319 @@
+Enumerations
+============
+
+The ``Enum`` class behaves differently from other Python classes in several 
+ways that require special-case handling in type checkers. This section discusses
+the Enum behaviors that should be supported by type checkers and others which
+may be supported optionally. It is recommended that library and type stub
+authors avoid using optional behaviors because these may not be supported
+by some type checkers.
+
+
+Enum Definition
+---------------
+
+Enum classes can be defined using a "class syntax" or a "function syntax".
+The function syntax offers several ways to specify enum members: names passed
+as individual arguments, a list of names, a tuple of names, or a string of
+comma-delimited names, or a string of space-delimited names.
+
+Type checkers should support the class syntax, but the function syntax (in
+is various forms) is optional::
+
+    class Color1(Enum): # Supported
+        RED = 1
+        GREEN = 2
+        BLUE = 3
+
+    Color2 = Enum('Color2', 'RED', 'GREEN', 'BLUE') # Optional
+    Color3 = Enum('Color3', ['RED', 'GREEN', 'BLUE']) # Optional
+    Color4 = Enum('Color4', ('RED', 'GREEN', 'BLUE')) # Optional
+    Color5 = Enum('Color5', 'RED, GREEN, BLUE') # Optional
+    Color6 = Enum('Color6', 'RED GREEN BLUE') # Optional
```

Comment:
Should these line comments follow this specification from pep 8?

> Inline comments should be separated by at least two spaces from the statement.

Most of the later line comments in this document follow this, but not all of them.